iron-form.html

```diff
@@ -456,6 +457,25 @@
 
         this.fire('iron-form-reset');
       }, 1);
+    },
+
+    /**
+     * Returns true if `node` is in the shadow DOM of a different element,
+     * that has also implemented IronFormElementBehavior and is registered
+     * to this form.
+     */
+    _isChildOfRegisteredParent: function(node) {
+      var parent = Polymer.dom(node).parentNode;
+
+      // At some point going up the tree we'll find either this form or the document.
+      while (parent && parent !== document && parent != this) {
+        // Use logical parentnode, or native ShadowRoot host.
+        parent = Polymer.dom(parent).parentNode || parent.host;
+        // Check if the parent was registered.
+        if (parent._parentForm === this)
```

Comment:
on the first loop, `parent = node.parentNode.parentNode` in this line. Doesn't seem correct, right?